pkg/config/apiserver.go

```diff
@@ -114,6 +114,52 @@ func (a *API) register(c *restful.Container) {
 		Reads(&pb.ServiceConfig{}).
 		Writes(APIResponse{}))
 
+	// Adapters
+	ws.Route(ws.
+		GET("/scopes/{scope}/adapters").
+		To(a.getAdaptersOrDescriptors).
+		Doc("Gets named adapter configurations.").
+		Param(ws.PathParameter("scope", "scope").DataType("string")).
+		Writes(APIResponse{}))
+
+	ws.Route(ws.
+		PUT("/scopes/{scope}/adapters").
+		To(a.putAdaptersOrDescriptors).
+		Doc("Creates or replaces named adapter configurations.").
+		Param(ws.PathParameter("scope", "scope").DataType("string")).
+		Reads(&pb.GlobalConfig{}).
+		Writes(APIResponse{}))
+
+	ws.Route(ws.
```

Comment:
In the global scope it is unwise to do.

However, if an operation fails validation it will not be allowed to proceed anyway.

Delete may be allowed in the future, so this returns a "Not implemented" vs a 404 which is confusing.